File: frontend/src/features/reservation/types/ReservationResponseDto.ts
import type { ReservedSeatDto } from '@/features/reservation/types/ReservedSeatDto';
import type { TrainDirection } from '@/features/schedule/types/TrainDirection';

export type ReservationResponseDto = {
    reservationId: string;
    scheduleCd: string;
    trainTypeName: string;
    departureTime: string;
    departureStationCd: string;
    departureStationName: string;
    arrivalTime: string;
    arrivalStationCd: string;
    arrivalStationName: string;
    rideDate: string;
    isDeleted: boolean;
    reservedSeats: ReservedSeatDto[];
    isReserverMatched: boolean;
    direction: TrainDirection;
};

File: backend/src/main/java/com/alab/shinkansendego/reservation/ReservationResponseDto.java
package com.alab.shinkansendego.reservation;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class ReservationResponseDto {
    private UUID reservationId;
    private String scheduleCd;
    private String trainTypeName;
    private String departureStationCd;
    private String departureStationName;
    private LocalTime departureTime;
    private String arrivalStationCd;
    private String arrivalStationName;
    private LocalTime arrivalTime;
    private LocalDate rideDate;
    private Boolean isDeleted;
    private List<ReservedSeatDto> reservedSeats;
    private Boolean isReserverMatched;
}
